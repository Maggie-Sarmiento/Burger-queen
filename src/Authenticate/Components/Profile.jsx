/* eslint-disable react/prop-types */
import * as React from 'react';
import { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { signOut } from 'firebase/auth';
import Button from '@mui/material/Button';
import Menu from '@mui/material/Menu';
import MenuItem from '@mui/material/MenuItem';
import Typography from '@mui/material/Typography';
import { auth } from '../../Firebase/firebase.config';
import style from './Profile.module.css';

const styles = {
  bgcolor: 'black',
  width: 60,
  height: 62,
  borderRadius: 60,
  color: '#F3B240',
  fontWeight: 'bold',
  fontSize: 20,
};

const Profile = ({ authenticate }) => {
  const [anchorEl, setAnchorEl] = useState(null);
  const open = Boolean(anchorEl);

  const navigate = useNavigate();

  const handleClose = () => {
    setAnchorEl(null);
  };

  const handleLogOut = async () => {
    await signOut(auth);
    handleClose();
    navigate('/');
  };

  const handleClick = (event) => {
    setAnchorEl(event.currentTarget);
  };

  return (
    <div>
      <Button
        className={style.btnProfile}
        id="basic-button"
        aria-controls={open ? 'basic-menu' : undefined}
        aria-haspopup="true"
        aria-expanded={open ? 'true' : undefined}
        onClick={handleClick}
        sx={styles}
      >
        GH
      </Button>
      <Menu
        id="basic-menu"
        anchorEl={anchorEl}
        open={open}
        onClose={handleClose}
        MenuListProps={{
          'aria-labelledby': 'basic-button',
        }}
      >
        <Typography id="modal-modal-title" variant="h6" component="h2">
          {authenticate.email}
        </Typography>
        <MenuItem onClick={handleLogOut}>Salir</MenuItem>
      </Menu>
    </div>
  );
};

export default Profile;
