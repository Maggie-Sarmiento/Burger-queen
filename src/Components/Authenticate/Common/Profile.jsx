/* eslint-disable react/prop-types */
import * as React from 'react';
import { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { signOut } from 'firebase/auth';
import { getDoc, doc } from 'firebase/firestore';
import Button from '@mui/material/Button';
import Menu from '@mui/material/Menu';
import MenuItem from '@mui/material/MenuItem';
import ExitToAppIcon from '@mui/icons-material/ExitToApp';
import Typography from '@mui/material/Typography';
import { auth, db } from '../../../Firebase/firebase.config';
import style from './Profile.module.css';

const styles = {
  bgcolor: 'black',
  minWidth: 0,
  width: 50,
  height: 52,
  borderRadius: 100,
  color: '#F3B240',
  fontWeight: 'bold',
  fontSize: 15,
};

const menu = {
  padding: 2,
  fontSize: '0.9em',
};
const btn = {
  display: 'flex',
  justifyContent: 'center',
  fontWeight: 'bold',
};

const Profile = ({ authenticate }) => {
  const [anchorEl, setAnchorEl] = useState(null);
  const [userName, setUserName] = useState(null);
  const [userRol, setUserRol] = useState(null);
  const [userLastName, setUserLastName] = useState(null);
  const open = Boolean(anchorEl);

  const user = authenticate;
  const getName = async (uid) => {
    const ref = doc(db, `profile/${uid}`);
    const docSnap = await getDoc(ref);
    const { name } = docSnap.data();
    const { lastname } = docSnap.data();
    const { role } = docSnap.data();
    setUserName(name);
    setUserLastName(lastname);
    setUserRol(role);
    return docSnap.data();
  };

  let letterName;
  let letterLastName;
  if (userName !== null) {
    letterName = userName.charAt(0).toUpperCase();
    letterLastName = userLastName.charAt(0).toUpperCase();
  }

  // const letterLastName = userLastName[0].toUpperCase();
  const userId = authenticate.uid;
  getName(userId);

  const navigate = useNavigate();

  const handleClose = () => {
    setAnchorEl(null);
  };

  const handleLogOut = async () => {
    await signOut(auth);
    handleClose();
    navigate('/');
  };

  const handleClick = (event) => {
    setAnchorEl(event.currentTarget);
  };

  return (
    <div>
      <Button
        className={style.btnProfile}
        id="basic-button"
        aria-controls={open ? 'basic-menu' : undefined}
        aria-haspopup="true"
        aria-expanded={open ? 'true' : undefined}
        onClick={handleClick}
        sx={styles}
      >
        {`${letterName}${letterLastName}`}
      </Button>
      <Menu
        id="basic-menu"
        anchorEl={anchorEl}
        open={open}
        onClose={handleClose}
        MenuListProps={{
          'aria-labelledby': 'basic-button',
        }}
      >
        <Typography sx={menu} id="modal-modal-title" variant="h6" component="h2">
          {userRol}
          <br />
          {`${userName} ${userLastName}`}
          <br />
          {user.email}
        </Typography>
        <MenuItem sx={btn} onClick={handleLogOut}>
          Salir&nbsp;
          <ExitToAppIcon />
        </MenuItem>
      </Menu>
    </div>
  );
};

export default Profile;
