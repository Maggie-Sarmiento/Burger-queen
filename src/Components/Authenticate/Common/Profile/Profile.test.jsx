/* eslint-disable no-undef */
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { create, act } from 'react-test-renderer';
import Profile from './Profile';

describe('test header', () => {
  test('render header', async () => {
    let root;
    const mockAuthenticate = jest.fn();
    act(() => {
      root = create(
        <Router>
          <Profile authenticate={mockAuthenticate} />
        </Router>,
      );
    });
    expect(root.toJSON()).toMatchSnapshot();
  });
});
