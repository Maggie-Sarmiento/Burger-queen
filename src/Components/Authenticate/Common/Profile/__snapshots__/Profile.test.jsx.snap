// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test header render header 1`] = `
<div>
  <button
    aria-haspopup="true"
    className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root btnProfile css-xa6ayv-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    id="basic-button"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    undefinedundefined
    <span
      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
